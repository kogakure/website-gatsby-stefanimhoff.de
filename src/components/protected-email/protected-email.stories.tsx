import React from 'react';

import { ProtectedEmail } from '.';

export default {
  component: ProtectedEmail,
  title: 'Helpers/ProtectedEmail',
};

export const Default = () => <ProtectedEmail />;

export const CustomTitle = () => <ProtectedEmail title="Email me today!" />;

export const NoKeyDownload = () => <ProtectedEmail noKeyDownload />;
