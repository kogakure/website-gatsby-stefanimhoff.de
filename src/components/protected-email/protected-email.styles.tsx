import styled from 'styled-components';

import { Lock } from '../../components/icons';

const ProtectedEmail = styled.span`
  background: transparent;
`;

const LockBox = styled.span`
  align-items: center;
  background: transparent;
  border: 0;
  cursor: pointer;
  display: inline-flex;
  height: 20px;
  justify-content: center;
  position: relative;
  top: 0.2em;
  width: 25px;
`;

const LockIcon = styled(Lock)`
  fill: var(--colorForeground);
  font-size: 1.2rem;
`;

export const Styled = {
  LockBox,
  LockIcon,
  ProtectedEmail,
};
