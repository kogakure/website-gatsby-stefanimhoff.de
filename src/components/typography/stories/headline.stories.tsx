import React from 'react';

import { Headline } from '..';

export default {
  title: 'Typography/Headline',
};

export const Default = () => <Headline>Headline</Headline>;

export const NoMargin = () => <Headline noMargin>Headline</Headline>;
