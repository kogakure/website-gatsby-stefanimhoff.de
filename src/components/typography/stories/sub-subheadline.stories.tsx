import React from 'react';

import { SubSubheadline } from '..';

export default {
  title: 'Typography/SubSubheadline',
};

export const Default = () => <SubSubheadline>SubSubheadline</SubSubheadline>;

export const NoMargin = () => (
  <SubSubheadline noMargin>SubSubheadline</SubSubheadline>
);
