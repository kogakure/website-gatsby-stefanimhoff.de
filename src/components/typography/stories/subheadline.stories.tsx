import React from 'react';

import { Subheadline } from '..';

export default {
  title: 'Typography/Subheadline',
};

export const Default = () => <Subheadline>Subheadline</Subheadline>;

export const NoMargin = () => <Subheadline noMargin>Subheadline</Subheadline>;
