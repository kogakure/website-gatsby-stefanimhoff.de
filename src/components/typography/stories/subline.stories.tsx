import React from 'react';

import { Subline } from '..';

export default {
  title: 'Typography/Subline',
};

export const Default = () => <Subline>Subline</Subline>;
