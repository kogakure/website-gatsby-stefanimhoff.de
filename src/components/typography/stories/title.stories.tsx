import React from 'react';

import { Title } from '..';

export default {
  title: 'Typography/Title',
};

export const Default = () => <Title>Title</Title>;

export const NoMargin = () => <Title noMargin>Title</Title>;
