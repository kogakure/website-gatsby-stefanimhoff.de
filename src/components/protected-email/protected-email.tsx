import * as React from 'react';

import { TextLink } from '../text-link';

import { Styled } from './protected-email.styles';

export type ProtectedEmailProps = {
  title?: string;
  noKeyDownload?: boolean;
};

export const ProtectedEmail: React.FC<ProtectedEmailProps> = ({
  noKeyDownload,
  title = 'Email',
  ...props
}) => {
  const [emailVisible, setEmailVisible] = React.useState(false);
  const downloadText = 'Download public key (ProtonMail/GPG)';

  const handleClick = (e: React.MouseEvent | React.KeyboardEvent) => {
    e.preventDefault();
    setEmailVisible(true);
  };

  return (
    <>
      {emailVisible ? (
        <Styled.ProtectedEmail>
          {!noKeyDownload && (
            <Styled.LockBox>
              <a
                title={downloadText}
                aria-label={downloadText}
                href="/downloads/publickey.stefan@imhoff.name-1d844c94d6fec7939a9bdaf221ce017ed5fa22c6.asc"
              >
                <Styled.LockIcon aria-hidden="true" />
              </a>
            </Styled.LockBox>
          )}
          <TextLink href="mailto:stefan@imhoff.name">
            stefan@imhoff.name
          </TextLink>
        </Styled.ProtectedEmail>
      ) : (
        <Styled.ProtectedEmail
          onClick={handleClick}
          onKeyUp={(e: React.KeyboardEvent) =>
            e.key === 'Enter' && handleClick(e)} // prettier-ignore
          role="button"
          tabIndex={0}
          {...props}
        >
          <TextLink href="#">{title}</TextLink>
        </Styled.ProtectedEmail>
      )}
    </>
  );
};
