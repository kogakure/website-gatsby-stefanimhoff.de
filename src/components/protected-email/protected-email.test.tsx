import React from 'react';
import { axe } from 'jest-axe';
import user from '@testing-library/user-event';

import { render, fireEvent } from '../../services/test-utils';

import { Default, CustomTitle, NoKeyDownload } from './protected-email.stories';

describe('ProtectedEmail', () => {
  test('renders correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('is accessible', async () => {
    const { container } = render(<Default />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });

  test('shows the email on click', async () => {
    const { container, findByText, getByRole } = render(<Default />);

    user.click(await findByText(/email/i));

    expect(container).toHaveTextContent(/stefan@imhoff.name/i);
    expect(getByRole('link', { name: /download public/i })).toBeInTheDocument();
  });

  test('show the email on Enter keyUp', async () => {
    const { container, findByText, getByRole } = render(<Default />);

    fireEvent.keyUp(await findByText(/email/i), {
      key: 'Enter',
      code: 13,
    });

    expect(container).toHaveTextContent(/stefan@imhoff.name/i);
    expect(getByRole('link', { name: /download public/i })).toBeInTheDocument();
  });

  describe('without GPG key download option', () => {
    test('renders correctly', () => {
      const { container } = render(<NoKeyDownload />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('shows the email on click', async () => {
      const { container, findByText, queryByRole } = render(<NoKeyDownload />);

      user.click(await findByText(/email/i));

      expect(container).toHaveTextContent(/stefan@imhoff.name/i);
      expect(
        queryByRole('link', { name: /download public/i })
      ).not.toBeInTheDocument();
    });

    test('show the email on Enter keyUp', async () => {
      const { container, findByText, queryByRole } = render(<NoKeyDownload />);

      fireEvent.keyUp(await findByText(/email/i), {
        key: 'Enter',
        code: 13,
      });

      expect(container).toHaveTextContent(/stefan@imhoff.name/i);
      expect(
        queryByRole('link', { name: /download public/i })
      ).not.toBeInTheDocument();
    });

    test('is accessible', async () => {
      const { container } = render(<NoKeyDownload />);
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });

  describe('with custom title', () => {
    test('renders correctly', () => {
      const { container } = render(<CustomTitle />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('is accessible', async () => {
      const { container } = render(<CustomTitle />);
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });
});
