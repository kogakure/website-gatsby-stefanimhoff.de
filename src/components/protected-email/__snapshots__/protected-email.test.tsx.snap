// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedEmail renders correctly 1`] = `
.c1 {
  color: var(--colorForeground);
  font-weight: 600;
  -webkit-text-decoration-color: var(--colorForegroundSoft);
  text-decoration-color: var(--colorForegroundSoft);
  -webkit-text-decoration-thickness: 0.1em;
  text-decoration-thickness: 0.1em;
  text-underline-offset: 0.1em;
}

.c1:hover,
.c1:focus {
  -webkit-text-decoration-color: var(--colorAccent);
  text-decoration-color: var(--colorAccent);
}

.c0 {
  background: transparent;
}

<span
  class="c0"
  role="button"
  tabindex="0"
>
  <a
    class="c1"
    href="#"
    rel="nofollow noopener noreferrer external"
  >
    Email
  </a>
</span>
`;

exports[`ProtectedEmail with custom title renders correctly 1`] = `
.c1 {
  color: var(--colorForeground);
  font-weight: 600;
  -webkit-text-decoration-color: var(--colorForegroundSoft);
  text-decoration-color: var(--colorForegroundSoft);
  -webkit-text-decoration-thickness: 0.1em;
  text-decoration-thickness: 0.1em;
  text-underline-offset: 0.1em;
}

.c1:hover,
.c1:focus {
  -webkit-text-decoration-color: var(--colorAccent);
  text-decoration-color: var(--colorAccent);
}

.c0 {
  background: transparent;
}

<span
  class="c0"
  role="button"
  tabindex="0"
>
  <a
    class="c1"
    href="#"
    rel="nofollow noopener noreferrer external"
  >
    Email me today!
  </a>
</span>
`;

exports[`ProtectedEmail without GPG key download option renders correctly 1`] = `
.c1 {
  color: var(--colorForeground);
  font-weight: 600;
  -webkit-text-decoration-color: var(--colorForegroundSoft);
  text-decoration-color: var(--colorForegroundSoft);
  -webkit-text-decoration-thickness: 0.1em;
  text-decoration-thickness: 0.1em;
  text-underline-offset: 0.1em;
}

.c1:hover,
.c1:focus {
  -webkit-text-decoration-color: var(--colorAccent);
  text-decoration-color: var(--colorAccent);
}

.c0 {
  background: transparent;
}

<span
  class="c0"
  role="button"
  tabindex="0"
>
  <a
    class="c1"
    href="#"
    rel="nofollow noopener noreferrer external"
  >
    Email
  </a>
</span>
`;
